test-harness.md

```diff
@@ -0,0 +1,94 @@
+MLS Interop Test Harness Specification
+======================================
+
+To automate interoperability testing across multiple MLS implementations, we
+need to be able to have a test runner control multiple clients, on different
+platforms and written in different languages. The interface to the MLS clients
+needs to cover all of the protocol operations for which we want to test interop,
+and expose enough information to evaluate whether interop succeeded.
+
+At a logical level, the necessary operations are all basically RPCs from the
+test runner to the clients.  First we describe the RPCs, then we discuss some
+ways these RPCs could be implemented.
+
+## Test Harness RPCs
+
+Ways to become a member of a group ("encrypt flag" indicates whether handshake
+messages in the group are encrypted):
+
+* Create a new group
+  * Inputs: Group ID, ciphersuite, credential, signature private key, encrypt
+    flag 
+  * Outputs: Pointer to new state
+* Create a new KeyPackage to begin a join
+  * Inputs: ciphersuite, credential, signature private key
+  * Outputs: KeyPackage, callback to which Welcome should be sent
+* Complete joining a group (Welcome callback)
+  * Inputs: Welcome, encrypt flag
+  * Outputs: Pointer to new state
+* External join
+  * Inputs: PublicGroupState, encrypt flag
+  * Outputs: MLSPlaintext(Commit), Welcome, pointer to new state
+
+Operations using a group state:
+
+* Get state auth
+  * Inputs: (none)
+  * Outputs: authentication_secret
+* Export a secret
+  * Inputs: label, context
```

Comment:
Is context the secret length?

Reply:
No, but we do need a length parameter.  The MLS exporter [takes three arguments](https://github.com/mlswg/mls-protocol/blob/master/draft-ietf-mls-protocol.md#exporters), label, context, and length.